### PR TITLE
BUG: Deconvolve method does not work with signals with more than 1-D …

### LINKED ISSUE #14997: BUG: Deconvolve method does not work with signals with more than 1-D
### Describe your issue.

I have an analogically filtered image and I would like to deconvolute it with the deconvolve method in scipy using a signal.butter filter that simulates my filter. So the image (the signal parameter) is not a 1-D array but a 2-D array and when passed as a parameter to the deconvolve method I get the underneath error. In the documentation, it is not written that the method only takes 1-D arrays as parameters but when looking down at the code we see that the signal is passed in the lfilter method that only takes 1-D arrays as parameters. I think that this is why we get the error. When creating the num and den variables (https://github.com/scipy/scipy/blob/7c904a77ad3d239bc35f18222810c9454e5441c3/scipy/signal/_signaltools.py#L2178), the atleast_1d method does not convert the parameters to 1-D arrays so that if we pass in the else condition, the lfilter method will get an error.
I guess, whether it should be added to the documentation that the only parameters accepted in the deconvolve method are 1-D arrays, whether the code should be corrected to convert the signal (and potentially the divisor too?) to a 1-D array before continuing the process.
It is my first contribution on GitHub and I'm really not an expert in Python nor in signal analysis so I hope this helps in any way.

### Reproducing Code Example

```python
import numpy as np
import matplotlib.pyplot as plt
from astropy.io import fits
from astropy.convolution import Gaussian2DKernel as gk
from astropy.convolution import convolve
from scipy import signal
from scipy.signal import butter

Gp = fits.open("C:/Users/.../.../.../.../.../jupiter_gprime2.fit")
dataGp = Gp[0].data

dataGCenter=dataGp[200:450,650:950]

# Simulating the filter used
filtering = signal.butter(2, [6.24*10**14,5.08*10**14], btype='bandpass', analog=True)

# Deconvolve the signal with the filter
signal.deconvolve(dataGCenter, filtering)
```

### Error message

```shell
File "C:\Users\...\anaconda3\lib\site-packages\scipy\signal\signaltools.py", line 1907, in 
lfilter
    return sigtools._linear_filter(b, a, x, axis)

ValueError: could not convert b, a, and x to a common type
```

### SciPy/NumPy/Python version information

1.5.2 1.19.2 sys.version_info(major=3, minor=8, micro=5, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by JuliaMelle01**
…#14997

Closes #14997

**What does it fix/implement?**
The error described in the issue is caused by the method lfilter(),
which is called in the method deconvolve().
This method is made for 1-D arrays as it is defined in the docstrings.
It follows that the deconcolve() method can also be used only for
1-D input arrays (filter and divisor). The deconvolve() method in
_signaltools.py was therefore slightly modified and two
tests (test_n_dimensional_signal(), test_n_dimensional_divisor()) were
added to the corresponding test class TestDeconvolve in test_signaltools.py.

The fix consists of:
1) a change in the docstring method description so that the input parameter
is explicitly referred to as a 1-D array instead of an array.
2) a simple if statement that throws a value error when a multidimensional
array is specified
3) two additional tests to verify that the fix works.

### CODE REVIEW

File: scipy/signal/_signaltools.py [hunk 1]
```diff
@@ -2142,18 +2142,18 @@ def deconvolve(signal, divisor):
...
-    signal : array_like
-        Signal data, typically a recorded signal
-    divisor : array_like
-        Divisor data, typically an impulse response or filter that was
-        applied to the original signal
+    signal : 1-D array_like
+    Signal data, typically a recorded signal
+    divisor : 1-D array_like
+    Divisor data, typically an impulse response or filter that was
+    applied to the original signal
```
Comment (tupui) [inline scipy/signal/_signaltools.py:2149]: Apart from the addition of 1-D, all docstring changes are not correct. Cf failing CI.

File: scipy/signal/_signaltools.py [hunk 2]
```diff
@@ -2167,16 +2167,20 @@ def deconvolve(signal, divisor):
...
-                    also accepts poly1d objects)
+    also accepts poly1d objects)
 
     """
     num = np.atleast_1d(signal)
     den = np.atleast_1d(divisor)
+    if num.ndim > 1:
+        raise ValueError("signal must be 1-D.")
+    if den.ndim > 1:
+        raise ValueError("divisor must be 1-D.")
```
Comment (tupui) [inline scipy/signal/_signaltools.py:2183]: I am not sure if we need 2 checks. I would have combined both. Let see what others say.

**Review by tupui — COMMENTED** [review]
Thanks @JuliaMelle01 for the PR! I made a few suggestions.

File: scipy/signal/_signaltools.py (hunk 2 above)
Comment (larsoner) [reply scipy/signal/_signaltools.py:2183]: This increases the specificity of the error, which is nice for the user. I'd keep them separate

Comment (JuliaMelle01) [reply scipy/signal/_signaltools.py:2183]: Thanks for the advices I intended the same as @larsoner says

File: scipy/signal/_signaltools.py [hunk 3]
```diff
@@ -2142,9 +2142,9 @@ def deconvolve(signal, divisor):
 
     Parameters
     ----------
-    signal : array_like
+    signal : 1-D array_like
```
Comment (tupui) [inline scipy/signal/_signaltools.py:2145]:
```suggestion
    signal : (N,) array_like
```

File: scipy/signal/_signaltools.py [hunk 4]
```diff
@@ -2142,9 +2142,9 @@ def deconvolve(signal, divisor):
...
         Signal data, typically a recorded signal
-    divisor : array_like
+    divisor : 1-D array_like
```
Comment (tupui) [inline scipy/signal/_signaltools.py:2147]:
```suggestion
    divisor : (N,) array_like
```

**Review by tupui — APPROVED** [review]
Thanks @JuliaMelle01. Sorry for the delay on this one.

**Pull request merged**